src/Features/Core/Portable/FindUsages/ExternalReferenceItem.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.CodeAnalysis.FindUsages
+{
+    /// <summary>
+    /// Information about a symbol's reference that can be used for display and navigation in an
+    /// editor.  These generally reference items outside of the Roslyn <see cref="Solution"/> model
+    /// provided by external sources (for example: RichNav).
+    /// </summary>
+    internal abstract class ExternalReferenceItem
+    {
+        /// <summary>
+        /// The definition this reference corresponds to.
+        /// </summary>
+        public DefinitionItem Definition { get; }
+
+        public ExternalReferenceItem(
+            DefinitionItem definition,
+            string documentName,
+            string projectName,
+            object text,
+            string displayPath)
+        {
+            Definition = definition;
+            DocumentName = documentName;
+            ProjectName = projectName;
+            Text = text;
+            DisplayPath = displayPath;
+        }
+
+        public string DocumentName { get; }
```

Comment:
This was deprecated in LocationWithText extension on VS LSP in lieu of DisplayPath (ie it can be retrieved from DisplayPath now via Path.GetFileName)

Reply:
ok.